src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -66,7 +66,7 @@ def __init__(self, metric_name: str, count: int, environment: uuid.UUID, grouped
             raise Exception(
                 'The character "%s" can not be used in the grouped_by value (%s)' % (METRIC_NAME_SEPARATOR, grouped_by)
             )
-        self.metric_name = ".".join([metric_name, grouped_by]) if grouped_by else metric_name
+        self.metric_name = METRIC_NAME_SEPARATOR.join([metric_name, grouped_by]) if grouped_by else metric_name
```

Comment:
Why do we use string concatenation between the name of a metric and the sub-category instead of using a separate field (both in the code as well as in the database). Wouldn't the latter make the code more easy to read?